src/Microsoft.Diagnostics.Runtime/src/Utilities/Platform/LinuxFunctions.cs

```diff
@@ -9,6 +9,59 @@ namespace Microsoft.Diagnostics.Runtime
 {
     internal sealed class LinuxFunctions : PlatformFunctions
     {
+        private const string LibDlGlibc = "libdl.so.2";
+        private const string LibDl = "libdl.so";
+
+        private readonly Func<string, IntPtr> _loadLibrary;
+        private readonly Func<IntPtr, bool> _freeLibrary;
+        private readonly Func<IntPtr, string, IntPtr> _getExport;
+
+        public LinuxFunctions()
```

Comment:
Use a static constructor?

Reply:
These are fine as instance.  For most of the library I do not like carrying state.